test/suite/lib.rs

```diff
@@ -63,74 +63,100 @@ async fn assert_balance(
     Ok(())
 }
 
+async fn setup_test_accounts(
+    ledger_env: &(impl LedgerEnv + LedgerTransaction + std::clone::Clone),
+    amount: Nat,
+    acc_num: u64,
+) -> anyhow::Result<Vec<impl LedgerEnv + LedgerTransaction + Clone>> {
+    let mut res = vec![];
+    assert!(acc_num > 0);
+    let balance = balance_of(ledger_env, ledger_env.principal()).await?;
+    assert!(balance >= amount.clone() + Nat::from(acc_num) * transfer_fee(ledger_env).await?);
+    for _i in 0..acc_num {
+        let receiver_env = ledger_env.fork();
+        let receiver = receiver_env.principal();
+        assert_balance(&receiver_env, receiver, 0).await?;
+
+        let _tx = transfer(ledger_env, Transfer::amount_to(amount.clone(), receiver))
+            .await
+            .with_context(|| format!("failed to transfer {} tokens to {}", amount, receiver))?;
+
+        assert_balance(
+            &receiver_env,
+            Account {
+                owner: receiver,
+                subaccount: None,
+            },
+            amount.clone(),
+        )
+        .await?;
+        res.push(receiver_env);
+    }
+    Ok(res)
+}
+
 /// Checks whether the ledger supports token transfers and handles
 /// default sub accounts correctly.
-pub async fn test_transfer(ledger_env: impl LedgerEnv + LedgerEnv) -> TestResult {
-    let receiver_env = ledger_env.fork();
-    let receiver = receiver_env.principal();
+/// Expects the given account to have a balance of at least 2*Transfer_Fee
+pub async fn test_transfer(ledger_env: impl LedgerEnv + LedgerTransaction + Clone) -> TestResult {
+    let envs = setup_test_accounts(&ledger_env, Nat::from(20_000), 2).await?;
```

Comment:
Nitpick: you don't need to clone. Convert `envs` into an iterator, make it mutable and then call `next` twice.